nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -75,21 +87,27 @@ protected String tableName() {
     return key.toString();
   }
 
-  @Override
-  protected void refreshFromMetadataLocation(String newLocation, Predicate<Exception> shouldRetry, int numRetries) {
-    super.refreshFromMetadataLocation(newLocation, shouldRetry, numRetries, this::loadTableMetadata);
-  }
-
-  private TableMetadata loadTableMetadata(String metadataLocation) {
+  private TableMetadata loadTableMetadata(String metadataLocation, Reference reference) {
     // Update the TableMetadata with the Content of NessieTableState.
-    TableMetadata.Builder builder = TableMetadata.buildFrom(TableMetadataParser.read(io(), metadataLocation))
+    TableMetadata deserialized;
+    if (table.getMetadata() != null) {
+      deserialized = TableMetadataParser.fromJson(io(), metadataLocation, table.getMetadata().getMetadata());
```

Comment:
So we have this path in to request metadata out of a Nessie specific object which stores the metadata as a Json String that needs to be parsed?

Reply:
(Not yet, but in the future, yes.)

It's already a JSON node tree (as a generic `JsonNode`) - so the text->nodes parsing already happened.
As I didn't want to serialize the `JsonNode` to a `String` to get a `TableMetadata`, I've updated the `TableMetadataParser`.